vignettes/questionnaires.Rmd

```diff
@@ -0,0 +1,627 @@
+---
+title: "Creating Questionnaire ADaMs"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Creating Questionnaire ADaMs}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes creating questionnaire ADaMs. Although questionnaire data
+is collected in a single SDTM dataset (`QS`), usually it does not make sense to
+create a single `ADQS` dataset for all questionnaire analyses. For example, a
+univariate analysis of scores by visit requires different variables than a
+time-to-event analysis. Therefore this vignette does not provide a programming
+workflow for a complete dataset but provides examples for deriving common types
+of questionnaire parameters.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless 
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(dplyr)
+```
+
+## Example Data
+
+In this vignette we use the example data from the CDISC ADaM Supplements
+([Generalized Anxiety Disorder 7-Item Version 2
+(GAD-7)](https://www.cdisc.org/standards/foundational/qrs/generalized-anxiety-disorder-7-item-version-2-0),
+[Geriatric Depression Scale Short Form
+(GDS-SF)](https://www.cdisc.org/standards/foundational/qrs/geriatric-depression-scale-short-form-0)):
+```{r echo=FALSE}
+suppress_warning(
+  qs_gad7 <- tibble::tribble(
+    ~QSSEQ, ~QSTESTCD, ~QSTEST, ~VISIT, ~VISITNUM, ~QSDTC, ~QSORRES,
+    1L, "GAD0201", "GAD02-Feeling Nervous Anxious or On Edge", "VISIT 1", 1, "2012-11-16", "More than half the days",
+    2L, "GAD0202", "GAD02-Not Able to Stop/Control Worrying", "VISIT 1", 1, "2012-11-16", "More than half the days",
+    3L, "GAD0203", "GAD02-Worrying Too Much About Things", "VISIT 1", 1, "2012-11-16", "More than half the days",
```

Comment:
Since this is small dataset should we just store in admiral.  That way folks don't have to go and retrieve from CDISC site?

Reply:
Moved to `example_script/example/qs.R` and added dataset to `data` folder.